### PR TITLE
Allow linking directly to specific tool and persist across site

### PR DESCRIPTION
**Pull request by DanTup**
@mit-mit @filiph Please lmk what you think.

I need to test in some other browsers (looks good on Chrome + Safari on my Mac). ~~I presume opening a PR will create a stage site so I easily test from others?~~

Changes:

1. When you switch tab, it will add that tool name to the url hash fragment
2. If you link to a page with a fragment (eg `#vscode`) it'll automatically select that tab
3. Your selected tab is written to localStorage and used (if there's no url fragment)

### CODE REVIEW

**Comment by DanTup** [comment]
https://dantup-flutter-staging.firebaseapp.com/

**Comment by DanTup** [comment]
Seems to work ok for me in Chrome (Mac+Win), Safari (Mac), IE and Edge but let me know if you can break anything.

**Comment by filiph** [comment]
That was fast! Could you please merge latest changes from master? We just launched beta and although I don't think I touched this code, I might have.

**Comment by DanTup** [comment]
@filiph Ok, rebased and updated staging:

https://dantup-flutter-staging.firebaseapp.com/

You might need to Ctrl+Refresh - I somehow ended up with the new site but with a sidebar, it was all a bit weird (I'm not sure how things are caching - hopefully users won't see that for the live site!).

**Comment by filiph** [comment]
LGTM. I'm asking @mit-mit for review as well.

File: js/tabs.js [hunk 1]
```diff
@@ -7,14 +8,36 @@ function setupToolsTabs() {
     tabContents.removeClass('current');
   }
 
+  function selectTool(id) {
```
Comment (filiph) [inline js/tabs.js]: Please add comments (what is the purpose of this function).

Comment (DanTup) [reply js/tabs.js]: Done! LMK if it's not clear enough.

**Comment by filiph** [comment]
paging @mit-mit for review, or pass

**Comment by mit-mit** [comment]
No comments from me (I'm pretty unknowledgeable about js), but sgtm for adding support for this

**Comment by DanTup** [comment]
@filiph LMK if you're happy for me to merge or want another set of eyes to check it.

**Comment by filiph** [comment]
Awesome, let's land this!

**Comment by DanTup** [comment]
Awesome, thanks! 👍 I'm not sure what the process is for deploying, but if you let me know when it's likely to be live, I'll give it a quick test.

**Comment by DanTup** [comment]
Oh, it's done straight from master via Travis. I'll test once it's finished :)

**Comment by DanTup** [comment]
Seems good to me (had to Ctrl+Refresh to get the new js though)

**Pull request merged**